src/routes/news/containers/WithCustomNewsProvider.js

```diff
@@ -0,0 +1,357 @@
+  }
+|}
+export type LoadingMoreType<S: {}> = {| status: "loadingMore", innerProps: S |}
+
+export type ErrorType<R: {}> = {|
+  status: "error",
+  message: ?string,
+  code: ErrorCodeType,
+  refreshProps: R | null
+|}
+
+export type ProviderPropType = {|
+  dispatch: Dispatch<StoreActionType>,
+  path: ?string,
+  selectedNewsType: string,
+  ...ProviderRefreshPropType
+|}
+
+export type LanguageNotAvailableType = {|
+  status: "languageNotAvailable",
+  availableLanguages: Array<LanguageModel>,
+  changeUnavailableLanguage: (
+    dispatch: Dispatch<StoreActionType>,
+    newLanguage: string
+  ) => void,
+  innerProps: {
+    path?: string,
+    selectedNewsType: string,
+    ...ProviderRefreshPropType
+  }
+|}
+
+export type SuccessType<S: {}, R: {}> = {|
+  status: "ready",
+  innerProps: S,
+  refreshProps: R
+|}
+
+export type StatusPropsType<S: {}, R: {}> =
+  | RouteNotInitializedType
+  | LoadingType
+  | LoadingMoreType<$Diff<S, { dispatch: Dispatch<StoreActionType> }>>
+  | ErrorType<R>
+  | LanguageNotAvailableType
+  | SuccessType<$Diff<S, { dispatch: Dispatch<StoreActionType> }>, R>
+
+export type PropsType<
+  S: {
+    ...ProviderPropType
+  },
+  R: {}
+> = {|
+  ...StatusPropsType<S, R>,
+  ...ProviderPropType
+|}
+
+const NewsTypeItem = ({ tab, onItemPress, selectedNewsType, t, theme }) => {
+  function onPress () {
+    onItemPress(tab.type)
+  }
+  const isLocal = tab.type === LOCAL
+  const isSelected = tab.type === selectedNewsType
+  return (
+    <TouchableWrapper onPress={onPress}>
+      {isLocal ? (
+        <LocalTabWrapper isSelected={isSelected} theme={theme}>
+          <LocalText theme={theme}>{t(tab.type)}</LocalText>
+        </LocalTabWrapper>
+      ) : (
+        <NewsTypeIcon source={isSelected ? tab.active : tab.inactive} />
+      )}
+    </TouchableWrapper>
+  )
+}
+
+const TranslatedNewsTypeItem = withTranslation('news')(
+  withTheme()(NewsTypeItem)
+)
+
+const withCustomNewsProvider = <
+  S: {
+    ...ProviderPropType
+  },
+  R: {}
+>(
+    refresh: (refreshProps: R, dispatch: Dispatch<StoreActionType>) => void
+  ): ((
+  Component: React.ComponentType<S>
+) => React.ComponentType<PropsType<S, R>>) => {
+  return (
+    Component: React.ComponentType<S>
+  ): React.ComponentType<PropsType<S, R>> => {
+    return class extends React.Component<
+      PropsType<S, R>,
+      {| selectedNewsType: string |}
+    > {
+      static displayName = wrapDisplayName(Component, 'withCustomNewsProvider');
+
+      state = {
+        selectedNewsType: this.getAvailableNewsType()
+      };
+
+      listRef = null;
+
+      getAvailableNewsType (): string {
+        const { cityModel, selectedNewsType } = this.props.innerProps || {}
+        const { tunewsEnabled, pushNotificationsEnabled } = cityModel || {}
+        let type = LOCAL
+        if (selectedNewsType) {
+          type = selectedNewsType
+        } else if (tunewsEnabled && !pushNotificationsEnabled) {
+          type = TUNEWS
```

Comment:
no need for another check `pushNotificationsEnabled && !tunewsEnabled`
using `else` is enough